lib/diff.ml

```diff
@@ -51,6 +62,15 @@ let modtype_item ~loc ~typing_env ~name ~reference ~current =
   | exception Includemod.Error _ ->
       Some (Module { mname = name; mdiff = Modified Unsupported })
 
+let type_item ~name ~reference ~current =
+  match (reference, current) with
+  | None, None -> None
+  | Some (Typ refType), None ->
+      Some (Type { tname = name; tdiff = Removed refType })
+  | None, Some (Typ curType) ->
+      Some (Type { tname = name; tdiff = Added curType })
+  | _ -> None
```

Comment:
You can make this last case more explicit, instead of using a wildcard. This makes it easier to read what case(s) remains, and lets OCaml's match exhaustiveness check tells you if you have forgotten to treat a case!